code/modules/mob/living/simple_animal/parrot.dm

```diff
@@ -787,6 +795,46 @@
 	src << "<span class='warning'>There is no perch nearby to sit on!</span>"
 	return
 
+
+/mob/living/simple_animal/parrot/proc/perch_mob_player()
+	set name = "Sit on Human's Shoulder"
+	set category = "Parrot"
+	set desc = "Sit on a nice comfy human being!"
+
+	if(stat || !client)
+		return
+
+	if(icon_state == "parrot_fly")
+		for(var/mob/living/carbon/human/H in view(src,1))
+			if(H.buckled_mob) //Already has a parrot, or is being eaten by a slime
+				continue
+			perch_on_human(H)
+			return
+		src << "<span class='warning'>There is nobody nearby that you can sit on!</span>"
+	else
+		icon_state = "parrot_fly"
+		parrot_state = PARROT_WANDER
+		if(buckled)
+			src << "<span class='notice'>You are no longer sitting on [buckled]'s shoulder.</span>"
+			buckled.buckled_mob = null
+		buckled = null
+		pixel_x = initial(pixel_x)
+		pixel_y = initial(pixel_y)
+
+
+
+/mob/living/simple_animal/parrot/proc/perch_on_human(mob/living/carbon/human/H)

```

Comment:
Please make this a bit more modular so I can add borg sitting more easily later! I would have to make a child of this  or name-change this proc in order to avoid copy paste.